### PR TITLE
[op-conductor] e2e tests - conductor rpc test fixes

### PR DESCRIPTION
**Pull request by henridevieux**
**Description**

e2e tests for conductor rpc. This fixes the issue that caused the revert https://github.com/ethereum-optimism/optimism/pull/9299

The specific commit: https://github.com/ethereum-optimism/optimism/commit/305acad49f37215137d8ebed5b0ac216907ef831 fixes the flaky test issue, failure reason is described below in comments

**Tests**

e2e tests for conductor rpc.

**Additional context**

Add any other context about the problem you're solving.

**Metadata**

- Fixes #[Link to Issue]
- [op-conductor] e2e tests  protocol-quest#85

### CODE REVIEW

**Review by zhwrd — APPROVED** [review]

File: op-e2e/sequencer_failover_setup.go [hunk 1]
```diff
@@ -299,24 +299,56 @@ func sequencerCfg(rpcPort int) *rollupNode.Config {
...
+		isLeader, err := prev.client.Leader(context.Background())
+		if err != nil {
+			return false, err
 		}
+		return !isLeader, nil
 	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	err := wait.For(ctx, 10*time.Second, condition)
```
Comment (0x00101010) [inline op-e2e/sequencer_failover_setup.go]: should we reduce the check interval from 10 to like 1?

File: op-e2e/sequencer_failover_setup.go [hunk 2]
```diff
@@ -299,24 +299,56 @@ func sequencerCfg(rpcPort int) *rollupNode.Config {
...
+		isActive, err := rollupClient.SequencerActive(context.Background())
+		if err != nil {
+			return false, err
+		}
+		return isActive == active, nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	return wait.For(ctx, 5*time.Second, condition)
```
Comment (0x00101010) [inline op-e2e/sequencer_failover_setup.go]: same here, reduce checking interval to speed up test case execution time

File: op-e2e/sequencer_failover_setup.go [hunk 3]
```diff
@@ -371,3 +403,38 @@ func findLeader(t *testing.T, conductors map[string]*conductor) (string, *conduc
...
+			if leader && active {
+				leaders++
+			}
+		}
+		return leaders == 1, nil
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	require.NoError(t, wait.For(ctx, 10*time.Second, condition))
```
Comment (0x00101010) [inline op-e2e/sequencer_failover_setup.go]: same here, reduce checking interval to speed up test case execution time.

These things typically happen within 1s

File: op-e2e/sequencer_failover_setup.go (hunk 1 above)
Comment (henridevieux) [reply op-e2e/sequencer_failover_setup.go]: Good call; lowered these

**Review by ajsutton — APPROVED** [review]
LGTM. We're likely to get port conflict issues though until we can enable startup retries in https://github.com/ethereum-optimism/optimism/pull/9543/

**Comment by henridevieux** [comment]
> LGTM. We're likely to get port conflict issues though until we can enable startup retries in #9543

Will take over that PR and update it for a fix

**Pull request merged**